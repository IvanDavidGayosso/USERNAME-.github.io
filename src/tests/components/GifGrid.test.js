import React from 'react'; 
import {shallow} from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs.js');

describe('Pruebas en <GifGrid />', () => {
    
    const category = 'Facebook';
    

    test('debe de mostrar el componente', () => {
        
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category }/>);
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar items cuando se carga imÃ¡genes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/example/img.jpm',
            title: 'img example',
        }];

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={category }/>);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
        
    });


    

})
