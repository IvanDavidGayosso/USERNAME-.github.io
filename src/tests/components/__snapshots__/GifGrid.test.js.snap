// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar el componente 1`] = `
<Fragment>
  <h3>
    Facebook
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se carga im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3>
    Facebook
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="img example"
      url="https://localhost/example/img.jpm"
    />
  </div>
</Fragment>
`;
